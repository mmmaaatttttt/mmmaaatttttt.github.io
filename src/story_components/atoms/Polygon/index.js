import React, { useMemo } from "react";
import PropTypes from "prop-types";
import COLORS from "utils/styles";

const defaultPoints = [
  { x: 0, y: 0 },
  { x: 100, y: 0 },
  { x: 100, y: 100 },
  { x: 0, y: 100 }
];
function Polygon({
  fill = COLORS.BLACK,
  open = false,
  points = defaultPoints,
  stroke = COLORS.BLACK,
  strokeWidth = 3
}) {
  const pointsStr = useMemo(() => points.map(p => `${p.x},${p.y}`).join(" "), [
    points
  ]);

  return open ? (
    <polyline
      points={pointsStr}
      fill={fill}
      stroke={stroke}
      strokeWidth={strokeWidth}
    />
  ) : (
    <polygon
      points={pointsStr}
      fill={fill}
      stroke={stroke}
      strokeWidth={strokeWidth}
    />
  );
}

Polygon.propTypes = {
  fill: PropTypes.string,
  open: PropTypes.bool,
  points: PropTypes.arrayOf(
    PropTypes.shape({
      x: PropTypes.number,
      y: PropTypes.number
    })
  ),
  stroke: PropTypes.string,
  strokeWidth: PropTypes.number
};

export default React.memo(Polygon);
