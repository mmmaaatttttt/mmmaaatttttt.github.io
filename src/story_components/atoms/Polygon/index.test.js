import React from "react";
import { render } from "@testing-library/react";
import Polygon from "./";

it("renders without crashing", () => {
  render(<Polygon />);
});

it("matches snapshot with props and children", () => {
  const { asFragment } = render(
    <Polygon
      fill="blue"
      stroke="purple"
      strokeWidth={1}
      points={[
        { x: 50, y: 50 },
        { x: 100, y: 200 },
        { x: 0, y: 200 }
      ]}
    >
      <div>hello</div>
      <div>goodbye</div>
    </Polygon>
  );
  expect(asFragment()).toMatchSnapshot();
});
