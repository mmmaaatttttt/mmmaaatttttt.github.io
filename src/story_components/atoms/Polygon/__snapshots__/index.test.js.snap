// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with props and children 1`] = `
<DocumentFragment>
  <polygon
    fill="blue"
    points="50,50 100,200 0,200"
    stroke="purple"
    stroke-width="1"
  />
</DocumentFragment>
`;
